src/ristretto/pedersen/extended_commitment_factory.rs

```diff
@@ -0,0 +1,451 @@
+        constants::{RISTRETTO_NUMS_POINTS, RISTRETTO_NUMS_POINTS_COMPRESSED},
+        pedersen::{
+            PedersenCommitment,
+            RISTRETTO_PEDERSEN_G,
+            RISTRETTO_PEDERSEN_G_COMPRESSED,
+            RISTRETTO_PEDERSEN_H,
+            RISTRETTO_PEDERSEN_H_COMPRESSED,
+        },
+        RistrettoPublicKey,
+        RistrettoSecretKey,
+    },
+};
+
+/// Generates extended Pederson commitments `sum(k_i.G_i) + v.H` using the provided base
+/// [RistrettoPoints](curve25519_dalek::ristretto::RistrettoPoints).
+/// Notes:
+///  - Homomorphism with public key only holds for extended commitments with `ExtensionDegree::Zero`
+#[derive(Debug, PartialEq, Clone)]
+pub struct ExtendedPedersenCommitmentFactory(pub(crate) PedersenGens<RistrettoPoint>);
+
+impl ExtendedPedersenCommitmentFactory {
+    /// Create a new Extended Pedersen Ristretto Commitment factory for the required extension degree using
+    /// pre-calculated compressed constants - we only hold references to the static generator points.
+    pub fn new_with_extension_degree(extension_degree: ExtensionDegree) -> Result<Self, RangeProofError> {
+        if extension_degree as usize > RISTRETTO_NUMS_POINTS.len() ||
+            extension_degree as usize > RISTRETTO_NUMS_POINTS_COMPRESSED.len()
+        {
+            return Err(RangeProofError::ExtensionDegree(
+                "Not enough Ristretto NUMS points to construct the extended commitment factory".to_string(),
+            ));
+        }
+        let mut g_base_vec = Vec::with_capacity(extension_degree as usize);
+        g_base_vec.push(RISTRETTO_PEDERSEN_G);
+        let mut g_base_compressed_vec = Vec::with_capacity(extension_degree as usize);
+        g_base_compressed_vec.push(RISTRETTO_PEDERSEN_G_COMPRESSED);
+        for i in 1..extension_degree as usize {
+            g_base_vec.push(RISTRETTO_NUMS_POINTS[i]);
+            g_base_compressed_vec.push(RISTRETTO_NUMS_POINTS_COMPRESSED[i]);
+        }
+        Ok(Self(PedersenGens {
+            h_base: *RISTRETTO_PEDERSEN_H,
+            h_base_compressed: *RISTRETTO_PEDERSEN_H_COMPRESSED,
+            g_base_vec,
+            g_base_compressed_vec,
+            extension_degree,
+        }))
+    }
+}
+
+impl Default for ExtendedPedersenCommitmentFactory {
+    /// The default Extended Pedersen Ristretto Commitment factory is of extension degree Zero; this corresponds to
+    /// the default non extended Pedersen Ristretto Commitment factory.
+    fn default() -> Self {
+        Self::new_with_extension_degree(ExtensionDegree::Zero).expect("Ristretto default base points not defined!")
+    }
+}
+
+impl ExtendedHomomorphicCommitmentFactory for ExtendedPedersenCommitmentFactory {
+    type P = RistrettoPublicKey;
+
+    fn commit(
+        &self,
+        k_i: &[RistrettoSecretKey],
+        v: &RistrettoSecretKey,
+    ) -> Result<PedersenCommitment, RangeProofError> {
+        let k_i: Vec<Scalar> = k_i.to_vec().iter().map(|k| k.0).collect();
```

Comment:
Unnecessary

Reply:
Logged the issue [here](https://github.com/tari-project/bulletproofs-plus/issues/9).